### PR TITLE
WebGPURenderer: support using 3d textures in shaders with texture3d() and add VolumeNodeMaterial() and examples.

### PR DESCRIPTION
**Pull request by aardgoose**
As title,

WebGPU and WebGL2 backends supported.
Note WGSL doesn't support a matrix inverse function so the inversion of the object's matrixWorld is carried out in JS.

Two examples based on WebGL2 examples: volume - cloud (simplified) and volume - perlin, both using the VolumeNodeMaterial().

(also Chrome release 125 now supports rendering to 3d textures. https://chromestatus.com/feature/5196871364771840 )

### CODE REVIEW

File: examples/jsm/nodes/accessors/Texture3DNode.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+		return normal( { texture: this, uv: uvNode } );
+
+	}
+
+}
+
+export default Texture3DNode;
+
+export const texture3D = nodeProxy( Texture3DNode );
```
Comment (RenaudRohlinger) [inline examples/jsm/nodes/accessors/Texture3DNode.js:98]: Can we also export `texture3D` in `Nodes.js` to expose it to userland? Also it seems that the lint was not applied for the `tslFn` normal function.

**Comment by RenaudRohlinger** [comment]
Overall, what do you think about letting the users handle the map properly themselves into the custom `testNode`?

For example something like this:
image

The `VolumeNodeMaterial` does not require a map by default to work, and it is used only within the custom `testNode` override function. This change would also allow for more complex custom testNode functions, UV mapping, and so on.

File: examples/jsm/nodes/accessors/Texture3DNode.js (hunk 1 above)
Comment (aardgoose) [reply examples/jsm/nodes/accessors/Texture3DNode.js:98]: exported texture3D and fixed the whitespace issues.

**Comment by aardgoose** [comment]
@sunag I had to add a hack in ReferenceNode after #28407 (before that PRm this hack wasn't needed) not sure what the correct solution is.



> Overall, what do you think about letting the users handle the map property themselves into the custom `testNode`?
> 
> For example something like this: image
> 
> The `VolumeNodeMaterial` does not require a map by default to work, and it is used only within the custom `testNode` override function. This change would also allow for more complex custom testNode functions, UV mapping, and so on. /cc @sunag

It works for me, initially I was just getting the functionality of the existing examples. Things like the base color should probably use materialColor etc to match existing materials.

**Comment by mrdoob** [comment]
I guess this PR should be updated now that #28433 got merged?

**Comment by aardgoose** [comment]
> I guess this PR should be updated now that #28433 got merged?

done.

File: examples/jsm/nodes/accessors/ReferenceNode.js [hunk 2]
```diff
@@ -90,6 +90,12 @@ class ReferenceNode extends Node {
 
 	getNodeType( builder ) {
 
+		if ( this.node === null ) {
+
+			return this.uniformType;
```
Comment (sunag) [inline examples/jsm/nodes/accessors/ReferenceNode.js:95]: Maybe we can call `this.updateValue()` here?

**Pull request merged**